docs/SQUASHING.md

```diff
@@ -1,16 +1,68 @@
 # How do I "squash" my commits?
 
-Before you start, [make sure your text editor is associated with Git](https://help.github.com/articles/associating-text-editors-with-git/) to not close without a response, otherwise you'll see some weird behavior.
+## Why do I need to "squash"?
 
-Then update your refs to be sure you have the latest code:
+Squashing is important to reduce the number of commits that go into develop and keep the history simpler. Often a feature branch may include a number of large commits as well as a few smaller ones for minor edits (like spelling and formatting fixes). This is good: you should always commit often on your feature branch. However, once everything has been hashed out and your Pull Request has been approved, we do not necessarily need the history of ever commit and it's respective message. For this reason, we like to "squash" our commits into one before merging a Pull Request. 
+
+The process of "squashing" your commits is done through a rebase.
+
+## What is a rebase?
+
+With large projects (such as ours), there are many different people working simultaneously. After submitting a pull request, chances are that someone else's pull request will be merged into develop before yours is, resulting in your feature branch being behind. When this happens, your branch goes "stale" and needs to be revitalized through a rebase. While stale, conflicts may occur if someone also edited a line you edited as well. Most of the time this is not the case, but either way, you will need to rebase before merging an approved pull request.
```

Comment:
- `develop`
- "yours" vs "yours is"
- remove "as well"
- remove last sentence